src/main/java/org/cbioportal/web/columnar/util/NewClinicalDataBinUtil.java

```diff
@@ -109,33 +98,48 @@ public static List<ClinicalDataBin> calculateDynamicDataBins(
         DataBinner dataBinner,
         List<ClinicalDataBinFilter> attributes,
         Map<String, ClinicalDataType> attributeDatatypeMap,
-        Map<String, List<Binnable>> filteredClinicalDataByAttributeId,
-        List<String> filteredUniqueSampleKeys,
-        List<String> filteredUniquePatientKeys
+        Map<String, List<Binnable>> filteredClinicalDataByAttributeId
     ) {
         List<ClinicalDataBin> clinicalDataBins = new ArrayList<>();
 
         for (ClinicalDataBinFilter attribute : attributes) {
 
             // if there is clinical data for requested attribute
             if (attributeDatatypeMap.containsKey(attribute.getAttributeId())) {
-                ClinicalDataType clinicalDataType = attributeDatatypeMap.get(attribute.getAttributeId());
-                List<String> filteredIds = clinicalDataType == ClinicalDataType.PATIENT
-                    ? filteredUniquePatientKeys
-                    : filteredUniqueSampleKeys;
-
                 List<ClinicalDataBin> dataBins = dataBinner
-                    .calculateDataBins(attribute, clinicalDataType,
-                        filteredClinicalDataByAttributeId.getOrDefault(attribute.getAttributeId(),
-                            emptyList()),
-                        filteredIds)
+                    .calculateDataBins(
+                        attribute,
+                        filteredClinicalDataByAttributeId.getOrDefault(attribute.getAttributeId(), emptyList())
+                    )
                     .stream()
                     .map(dataBin -> dataBinToClinicalDataBin(attribute, dataBin))
-                    .collect(toList());
+                    .toList();
                 clinicalDataBins.addAll(dataBins);
             }
         }
 
         return clinicalDataBins;
     }
+
+    /**
```

Comment:
comment on why we are doing this kinda strange operation (for optimization and to leverage legacy bin counting logic)